### DISCUSSION TITLE
Proposal to include soft links when listing the available models

### DISCUSSION THREAD

**Discussion by CalculonPrime** [vladmandic/sdnext#3607] [Ideas]
For automatic1111, I set up an optimized layout for downloaded models that leverages that web UI's ability to pick up soft links:

- I have the models directory pointing at my D drive, which is a large but slow disk (rotating storage).  The stable diffusion models thus have the path D:\models\Stable-diffusion\<model file name>.  This is by default where I store model files, especially new ones.
- On my C drive, which is a NVME solid state drive, faster but smaller, I have created a parallel directory C:\models\Stable-diffusion.  When a model becomes one of my favorites, I promote it to this drive, and create a soft link to it from the D drive, so it appears there as well.

Automatic1111 shows all models with this setup, both actual files and soft links..  My proposal is that you add similar support for soft links so I can use this same approach with your web UI.  Note that I can't create hard links between drives, so that's not a solution.

I suspect this just requires changing the flags used when fetching the list of model files for stable diffusion.  If you aren't interested in implementing this, would you accept a file change from me doing it?

**Comment by vladmandic** [comment]
> If you aren't interested in implementing this, would you accept a file change from me doing it?

honestly, i don't want to go through setup of validation environment.
if you create a pr and test it, i'm not against it.
